lib/http/intercepted_client.dart

```diff
@@ -264,7 +270,11 @@ class InterceptedClient extends BaseClient {
 
       var stream = requestTimeout == null
           ? await _inner.send(interceptedRequest)
-          : await _inner.send(interceptedRequest).timeout(requestTimeout!);
+          : await _inner.send(interceptedRequest).timeout(requestTimeout!,
+              onTimeout: () async {
+              if (onRequestTimeout != null) onRequestTimeout!();
```

Comment:
since this function is optional, I think it would improve if the call is made without and if and instead used Dart's nullable checks. It would look something like this:

`onRequestTimeout?.call()`